app/listings/[id]/actions.ts

```
@@ -0,0 +1,57 @@
+
+  if (error) {
+    console.error("Error flagging user:", error);
+    return { error: "Failed to report user." };
+  }
+
+  // Optional: You could also create a notification for admins here
+
+  revalidatePath(`/listings/${listingId}`);
+
+  // Track event with PostHog
+  const posthog = new PostHog(process.env.NEXT_PUBLIC_POSTHOG_KEY!)
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)